spec/services/oauth_clients/one_drive_connection_manager_spec.rb

```diff
@@ -39,6 +39,55 @@
     described_class.new(user:, configuration: storage.oauth_configuration)
   end
 
+  describe '#code_to_token' do
+    let(:code) { 'wow.such.code.much.token' }
+    let(:code_to_token_response) do
+      {
+        access_token: "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ik5HVEZ2ZEstZnl0aEV1Q...",
+        token_type: "Bearer",
+        expires_in: 3599,
+        scope: "Mail.Read User.Read",
+        refresh_token: "AwABAAAAvPM1KaPlrEqdFSBzjqfTGAMxZGUTdM0t4B4..."
+      }.to_json
+    end
+    let(:me_response) do
+      {
+        businessPhones: [
+          "+45 123 4567 8901"
+        ],
+        displayName: "Sheev Palpatine ",
+        givenName: "Sheev",
+        jobTitle: "Galatic Senator",
+        mail: "palpatine@senate.com",
+        mobilePhone: "+45 123 4567 8901",
+        officeLocation: "500 Republica",
+        preferredLanguage: "en-US",
+        surname: "Palpatine",
+        userPrincipalName: "palpatine@senate.com",
+        id: "87d349ed-44d7-43e1-9a83-5f2406dee5bd"
+      }.to_json
+    end
+
+    before do
+      stub_request(:post, 'https://login.microsoftonline.com/consumers/oauth2/v2.0/token')
+        .to_return(status: 200, body: code_to_token_response, headers: { 'Content-Type' => 'application/json' })
+
+      stub_request(:get, 'https://graph.microsoft.com/v1.0/me')
+        .with(headers: { Authorization: "Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ik5HVEZ2ZEstZnl0aEV1Q..." })
+        .to_return(status: 200, body: me_response, headers: { 'Content-Type' => 'application/json' })
```

Comment:
🟡 how does it behave, if request for user id fails (e.g. timeout). Can we have a test case for that?

Reply:
Raise for status gives us an exception for any non status greater than 299.

Reply:
Added a test case for the raising of the error. Again, should not happen, but better safe than sorry.

Reply:
yes, that's adequate. go ahead with merging! 🟢